_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+| **push** to `alpha`                                 | Get the last release for commits on branch `alpha`; if the last release is also present on a lower branch (it would be the first release on `alpha`), increment the version based on new commits and make a prelease on `alpha` channel; if the last release is a pre-release increment the prerelease version (`5.0.0-alpha.1`, then `5.0.0-alpha.2`) and release on `alpha` channel |
+| **merge** any branch to `alpha`                     | Get the last release for commits on branch `alpha`; if the last release is also present on a lower branch (it would be the first release on `alpha`), increment the version based on new commits and make a prelease on `alpha` channel; if the last release is a pre-release increment the prerelease version (`5.0.0-alpha.1`, then `5.0.0-alpha.2`) and release on `alpha` channel |
+
+##### Pre-releases configuration
+
+```json
+{
+  "release": {
+    "branches": [
+      "master",
+      {"branch": "beta", "prerelease": true},
+      {"branch": "alpha", "prerelease": true},
+    ]
+  }
+}
+```
+
+#### Configuration
+
+##### Format
+
+Each branch is either a `String` or an `Object` with the following properties:
+
+| Option       | Description                                                                                                                                                             |
+|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `branch`     | **Require** Git branch name                                                                                                                                             |
+| `range`      | Accepted version range for to release from this branch. If defined the branch will be considered a LTS branch.                                                          |
+| `channel`    | Channel on which to release. Ignored for release targets that don't support channels.                                                                                   |
+| `prerelease` | Version of the pre-release done from this branch. Must be formatted `<version>-<tag>`. The version is required, the `tag` is optional. Examples: `4.0.0-beta`, `4.0.0`. |
+
+**The order in which the branches are defines is meaningful, and it is used to determine the branch type and which release can be done from each branch.**
+
+##### Branches naming convention
+
+In a branch is defined as a `String` `semantic-release` will automatically determine the type of branch as follow:
+- If the branch name is a valid semver range (Only the format `1.x.x` or `1.x` are valid Git reference names) the branch will be considered a LTS branch, with its range based on the branch name, and release on the channel with the same name
```

Comment:
I would not mention `1.x.x`, just the `1.x` format

Reply:
Several project maintain both breaking and feature legacy branches.
It's quite common to release a fix that applies to a specific feature from a legacy release.

For example if you discover a bug in version 2.1.1, and some users don't want to upgrade to 2.2.x, nor 3.x.x.
That's not a common situation in OSS, but in corporate world it's very, very common, because moving from a 2.1.x release to 2.2.x would trigger a whole process of paperwork, management approval, integration tests, user acceptance tests etc...
Applying patches on legacy feature versions is 80+ percent of releases I do in my day job (large corporation world).

So in such case one could branch off `master` on tag 2.1.1 and create the branch `2.1.x`, make the fix and that would release a 2.1.2 that customer can install quickly as it would be considered "low risk".

Also as mentioned in previous comment, implementing that in a generic way using semver would result in simpler code and avoid to add extra complexity by hard-coding limitations.

Reply:
Okay, glad you have these enterprise insights because I heard that before :) I trust you on this, lets keep it as you suggest.

Reply:
> Several project maintain both breaking and feature legacy branches.

I've seen this quite often as well. Application teams prefer to remain on a given `minor` release until a `minor` upgrade can be thoroughly vetted by QA (Given the concern that a `minor`, though not technically a breaking release, may contain unintended breakage resulting from underlying code refactoring). In the meantime, they still expect bug fixes in newer releases to be back-ported to the `minor` version they are already using.